src/python_testing/mdns_discovery/mdns_discovery.py

```diff
@@ -0,0 +1,275 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+
+import asyncio
+from typing import Dict, List
+
+from mdns_discovery.exceptions import DiscoveryNotPerformedError
+from zeroconf import IPVersion, ServiceStateChange, Zeroconf
+from zeroconf.asyncio import AsyncServiceBrowser, AsyncServiceInfo
+
+from .mdns_type_enum import MdnsType
+
+
+class MdnsDiscovery:
+
+    DEFAULT_DISCOVERY_DURATION_SEC = 3
+
+    def __init__(self, tc):
+        """
+        Initializes the MdnsDiscovery instance with necessary configurations.
+
+        Args:
+            tc: A test case object containing node and fabric information.
+
+        Attributes:
+            _zc (Zeroconf): An instance of Zeroconf to manage mDNS operations.
+            _aiobrowser: An instance of the AsyncServiceBrowser browser to discover services.
+            _tc: The test case object.
+            _service_types (list): A list of mDNS service types to discover.
+            _discovered_services (dict): A dictionary to store discovered services.
+            _discovery_performed (bool): Flag to indicate if discovery has been performed.
```

Comment:
Do not expose private members in documentation., just document them inline of the __init__ where they are declared.